src/kubernetes.cc

```diff
@@ -102,6 +109,68 @@ bool ReadServiceAccountSecret(
   return !input.fail();
 }
 
+// Extracts the API group and version from the api_version.
+const std::pair<std::string, std::string> ApiGroupAndVersion(
+    const std::string& api_version) {
+  std::vector<std::string> slash_split;
+  boost::algorithm::split(
+      slash_split, api_version, boost::algorithm::is_any_of("/"));
+  if (slash_split.size() == 1) {
+    // api_version has the format "<version>". E.g. "v1".
+    return std::make_pair(std::string(kK8sCoreGroup), slash_split[0]);
+  } else if (slash_split.size() == 2) {
+    // api_version has the format "<group-name>/<version>". E.g. "apps/v1".
+    return std::make_pair(slash_split[0], slash_split[1]);
+  } else {
+    LOG(ERROR) << "Invalid apiVersion: " << api_version;
+  }
+}
+
+// Builds the resource type and version of a Kubernetes resource given its API
+// version and kind.
+const std::pair<std::string, std::string> TypeAndVersion(
+    const std::string& api_version, const std::string& kind) {
+  const std::pair<std::string, std::string> group_and_version =
+      ApiGroupAndVersion(api_version);
+  const std::string group_name = group_and_version.first;
+  std::string resource_type;
+  if (group_name == kK8sCoreGroup) {
+    resource_type =
+        format::Substitute(kK8sCoreGroupResourceTypeFormat, {{"kind", kind}});
+  } else {
+    const std::string resource_type =
+        format::Substitute(kK8sNamedGroupsResourceTypeFormat,
+                           {{"group_name", group_name}, {"kind", kind}});
+  }
+  return std::make_pair(resource_type, group_and_version.second);
+}
+
+// Build the watch path given the Kubernetes resource kind and API version.
+const std::string WatchPath(
+    const std::string& plural_kind, const std::string& api_version,
+    const std::string& selector) {
+  const std::pair<std::string, std::string> group_and_version =
+      ApiGroupAndVersion(api_version);
+  const std::string group_name = group_and_version.first;
+  const std::string version = group_and_version.second;
+  std::string api_path;
+  if (group_name == kK8sCoreGroup) {
```

Comment:
Here you don't even need a `slash_split` — just test whether `api_version` contains a `'/'`.